internal/circle/client.go

```diff
@@ -23,7 +23,7 @@ const (
 
 // ClientInterface defines the interface for interacting with the Circle API.
 //
-//go:generate mockery --name=ClientInterface --case=underscore --structname=MockClient
+//go:generate mockery --name=ClientInterface --case=underscore --structname=MockClient --filename=client_mock.go --inpackage
```

Comment:
This change moved the circle mocks from `.../circle/mocks` to `.../circle`, which was needed in order to allow these mocks to be used within the Circle package without falling into circular dependency.

The circular dependency was caused by many of the circle structs being needed for these mocks.